### PR TITLE
Fix JavaFX thread exception when fetching new infos

### PR DESCRIPTION
**Pull request by tobiasdiez**
Fixes JavaFX thread exceptions by reworking the SwingWorker using JavaFX BackgroundTasks.

### CODE REVIEW

File: src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java [hunk 1]
```diff
@@ -41,11 +54,41 @@ public FetchAndMergeEntry(BibEntry entry, String field) {
...
-            if (entry.hasField(field)) {
-                new FetchAndMergeWorker(panel, entry, field).execute();
+            Optional<String> fieldContent = entry.getField(field);
+            if (fieldContent.isPresent()) {
+                BackgroundTask.wrap(() -> {
+                    Optional<IdBasedFetcher> fetcher = WebFetchers.getIdBasedFetcherForField(field, Globals.prefs.getImportFormatPreferences());
+                    if (fetcher.isPresent()) {
+                        return fetcher.get().performSearchById(fieldContent.get());
+                    } else {
+                        return Optional.<BibEntry>empty();
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java]: Normally you don't need the genericc type in return Optional.empty

File: src/main/java/org/jabref/gui/entryeditor/EntryEditor.java [hunk 2]
```diff
@@ -353,6 +358,24 @@ private void setupToolBar() {
...
+                      .onSuccess(fetchedEntry -> {
+                          if (fetchedEntry.isPresent()) {
+                              MergeFetchedEntryDialog dialog = new MergeFetchedEntryDialog(panel, entry, fetchedEntry.get(), fetcher.getName());
+                              dialog.setVisible(true);
+                          } else {
+                              dialogService.notify(Localization.lang("Could not find any bibliographic information."));
+                          }
+                      })
+                      .onFailure(exception -> {
+                          LOGGER.error("Error while fetching entry with " + fetcher.getName(), exception);
```
[the remark this answers is not in the document]
Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/entryeditor/EntryEditor.java]: This is not possible if the last argument should be treated as an exception.

File: src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java (hunk 1 above)
Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java]: I was also surprised but it leads to compiler errors without it.

File: src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java [hunk 3]
```diff
@@ -41,11 +55,41 @@ public FetchAndMergeEntry(BibEntry entry, String field) {
      * @param fields - List of fields to get information from, one at a time in given order
      */
     public FetchAndMergeEntry(BibEntry entry, BasePanel panel, List<String> fields) {
+        this.dialogService = panel.frame().getDialogService();
+        this.panel = panel;
+
+        // TODO: Don't run this method as part of the constructor
```
Comment (LinusDietz) [inline src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java]: Can you resolve this TODO?

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java]: Yeah, I was just too lazy to do it ;-). Fixed now.

**Pull request merged**